morph/readability.py

```diff
@@ -0,0 +1,369 @@
+            except:
+                self.writeOutput("Failed to process '%s'\n" % file_name)
+                raise
+
+        def accepted_filetype(filename):
+            lower_filename = filename.lower()
+            if lower_filename.endswith('.srt'):
+                return True
+            if lower_filename.endswith('.ass'):
+                return True
+            if lower_filename.endswith('.txt'):
+                return True
+            return False
+        
+        list_of_files = list()
+        for (dirpath, _, filenames) in os.walk(input_path):
+            list_of_files += [os.path.join(dirpath, filename) for filename in filenames if accepted_filetype(filename)]
+
+        if len(list_of_files) > 0:
+            self.writeOutput("\n[Current Readability]\n")
+            self.writeOutput('%s\t%s\t%s\t%s\t%s\t%s\t%s\n' % ("Input", "Total Morphs", "Known Morphs", "% Known Morphs", "Total Instances", "Known Instances", "% Readability"))
+
+            mw.progress.start( label='Updating data', max=len(list_of_files), immediate=True )
+            for n, file_path in enumerate(sorted(list_of_files, key=natural_keys)):
+                mw.progress.update( value=n, label='Parsing (%d/%d) %s' % (n+1, len(list_of_files), os.path.basename(file_path)) )
+                if os.path.isfile(file_path):
+                    is_ass = os.path.splitext(file_path)[1].lower() == '.ass'
+                    is_srt = os.path.splitext(file_path)[1].lower() == '.srt'
+                    measure_readability(file_path, is_ass, is_srt)
+            mw.progress.finish()
+        else:
+            self.writeOutput('\nNo files found to process.\n')
+
+        if save_word_report:
+            self.writeOutput("\n[Saving word report to '%s'...]\n" %  word_report_path)
+            with open(word_report_path, 'wt', encoding='utf-8') as f:
+                last_count = 0
+                morph_idx = 0
+                group_idx = 0
+                morph_total = 0.0
+                all_morphs_count = sum(n for n in all_morphs.values())
+
+                for m in sorted(all_morphs.items(), key=operator.itemgetter(1), reverse = True):
+                    if m[1] != last_count:
+                        last_count = m[1]
+                        group_idx += 1
+                    morph_idx += 1
+                    morph_delta = 100.0 * m[1] / all_morphs_count
+                    morph_total += morph_delta
+                    part0 = '*' if m[0].pos == '' else m[0].pos
+                    part1 = '*' if m[0].subPos == '' else m[0].subPos
+                    print('%d\t%s\t%d\t%d\t%0.8f\t%0.8f\t%s' % (m[1], m[0].base, group_idx, morph_idx, morph_delta, morph_total, part0 + ',' + part1 + ',*,*'), file=f)
+
+        
+
+        learned_tot = 0
+        learned_morphs = []
+
+        all_missing_morphs = []
+
+        if save_study_plan:
+            self.writeOutput("\n[Saving Sutudy Plan to '%s'...]\n" %  study_plan_path)
```

Comment:
Thanks, I'll take a look at the UI cut-off.

The columns are tab-separated so they can be copy-pasted into a spreadsheet.  Aligning them would break that, and aligning text with Japanese characters is tricky (they are not the same width as western text).  We could change this to a QTableWidget at some point.